File: stripe/src/main/java/com/stripe/android/ApiResultCallback.java
package com.stripe.android;

import androidx.annotation.NonNull;

/**
 * Generic interface for an API operation callback that either returns a
 * result, {@link ResultType}, or an {@link Exception}
 */
public interface ApiResultCallback<ResultType> {
    void onSuccess(@NonNull ResultType result);

    void onError(@NonNull Exception e);
}

File: payments-core/src/main/java/com/stripe/android/ApiResultCallback.kt
package com.stripe.android

import com.stripe.android.core.model.StripeModel

/**
 * Generic interface for an API operation callback that either returns a
 * result, [ResultType], or an [Exception]
 */
interface ApiResultCallback<in ResultType : StripeModel> {
    fun onSuccess(result: ResultType)

    fun onError(e: Exception)
}
